detox/src/devices/common/drivers/android/tools/HashHelper.js

```diff
@@ -0,0 +1,46 @@
+const crypto = require('crypto');
+const { DetoxRuntimeError } = require('../../../../../errors');
+
+const SUPPORTED_HASH_TYPES = {
+  MD5: 'md5',
+}
+
+const HASH_PATH = '/data/local/tmp/detox';
+
+class HashHelper {
+  constructor(adb, hashXfer, hashPath = HASH_PATH) {
+    this._adb = adb;
+    this._hashXfer  = hashXfer;
+    this._hashPath = hashPath;
+  }
+
+  generateHash(path, hashType = SUPPORTED_HASH_TYPES.MD5) {
+    if (!path) {
+      throw new DetoxRuntimeError({
+        message: `Path must be provided for hash generation`,
+      });
+    }
+
+    switch (hashType) {
+      case SUPPORTED_HASH_TYPES.MD5:
+        return crypto.createHash('md5').update(path).digest("hex");
```

Comment:
Why do you calculate an MD5 hash of a *path*, not the file contents themselves? 🤔